### DISCUSSION TITLE
Creating a Cookie based authentication

### DISCUSSION THREAD

**Discussion by iudeen** [fastapi/fastapi#6751] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

I am looking to integrate Cookie based authentication in my FastAPI App. I want the same to work seamlessly with swagger as well. 

I want to have a route (eg: `/login`) which sets my browser cookies. All other protected route uses `Depends` in the decorator to verify the key present in cookie. How do I get this to work with OpenAPI authorize button?

**Comment by alikazai** [comment]
hi have a look at this https://medium.com/data-rebels/fastapi-how-to-add-basic-and-cookie-authentication-a45c85ef47d3
hope it helps.

**Comment by iudeen** [comment]
> hi have a look at this https://medium.com/data-rebels/fastapi-how-to-add-basic-and-cookie-authentication-a45c85ef47d3
> hope it helps.

Hey,

Thanks for this. I will try this out and give an update.

**Comment by btall** [comment]
Hi 👋🏾   
In a slightly broader scope, you can also use *FastAPI-Users* for what you want to do. https://frankie567.github.io/fastapi-users/
